Source/WebCore/Modules/cookie-store/CookieStore.cpp

```diff
@@ -56,25 +65,243 @@ namespace WebCore {
+                if (!cookies)
+                    completionHandler(Exception { TypeError });
+                else
+                    completionHandler(WTFMove(*cookies));
+            });
+        };
+
+        cookieJar.getCookiesAsync(document, document->url(), options, WTFMove(resultHandler));
+    };
+
+    ensureOnMainThread(WTFMove(getCookies));
+}
+
+void CookieStore::MainThreadBridge::getAll(CookieStoreGetOptions&& options, RefPtr<SecurityOrigin> origin, bool globalObjectIsWindowObject, CompletionHandler<void(ExceptionOr<Vector<Cookie>>&&)>&& completionHandler)
+{
+    auto getAllCookies = [this, protectedThis = Ref { *this }, options = WTFMove(options), origin, globalObjectIsWindowObject, completionHandler = WTFMove(completionHandler)](ScriptExecutionContext& context) mutable {
+        Ref document = downcast<Document>(context);
+        WeakPtr page = document->page();
+        if (!page) {
+            ensureOnContextThread([completionHandler = WTFMove(completionHandler)]() mutable {
+                completionHandler(Exception { SecurityError });
+            });
+            return;
+        }
+
+        auto url = document->url();
+        if (!options.url.isNull()) {
+            auto parsed = document->completeURL(options.url);
+            if (globalObjectIsWindowObject && parsed != url) {
+                ensureOnContextThread([completionHandler = WTFMove(completionHandler)]() mutable {
+                    completionHandler(Exception { TypeError });
+                });
+                return;
+            }
+
+            if (!origin->isSameOriginDomain(SecurityOrigin::create(parsed))) {
+                ensureOnContextThread([completionHandler = WTFMove(completionHandler)]() mutable {
+                    completionHandler(Exception { TypeError });
+                });
+                return;
+            }
+            url = WTFMove(parsed);
+        }
+
+        auto& cookieJar = page->cookieJar();
+        auto resultHandler = [this, protectedThis = WTFMove(protectedThis), completionHandler = WTFMove(completionHandler)] (std::optional<Vector<Cookie>>&& cookies) mutable {
+            ensureOnContextThread([completionHandler = WTFMove(completionHandler), cookies = WTFMove(cookies)]() mutable {
+                if (!cookies)
+                    completionHandler(Exception { TypeError });
+                else
+                    completionHandler(WTFMove(*cookies));
+            });
+        };
+
+        cookieJar.getCookiesAsync(document, url, options, WTFMove(resultHandler));
+    };
+
+    ensureOnMainThread(WTFMove(getAllCookies));
+}
+
+void CookieStore::MainThreadBridge::set(CookieInit&& options, CompletionHandler<void(ExceptionOr<void>&&)>&& completionHandler)
+{
+    auto setCookie = [this, protectedThis = Ref { *this }, options = crossThreadCopy(WTFMove(options)), completionHandler = WTFMove(completionHandler)](ScriptExecutionContext& context) mutable {
```

Comment:
Funny how you property called crossThreadCopy() in only one of your 3 functions.